code/lib/cli/src/automigrate/fixes/new-frameworks.ts

```diff
@@ -0,0 +1,235 @@
+      logger.warn('Unable to find storybook main.js config, skipping');
+      return null;
+    }
+
+    const storybookCoerced = storybookVersion && semver.coerce(storybookVersion)?.version;
+    if (!storybookCoerced) {
+      logger.warn(dedent`
+        ❌ Unable to determine storybook version, skipping ${chalk.cyan('newFrameworks')} fix.
+        🤔 Are you running automigrate from your project directory?
+      `);
+      return null;
+    }
+
+    if (!semver.gte(storybookCoerced, '7.0.0')) {
+      console.log('lower than 7.0.0!');
+      return null;
+    }
+
+    // If in the future the eslint plugin has a framework option, using main to extract the framework field will be very useful
+    const main = await readConfig(mainConfig);
+
+    if (!main) {
+      console.log('no main');
+      return null;
+    }
+
+    const frameworkPackage = main.getFieldValue(['framework']) as keyof typeof packagesMap;
+    const builder = main.getFieldValue(['core', 'builder']);
+
+    if (!frameworkPackage || !builder) {
+      console.log('no framework or no builder, skipping');
+      return null;
+    }
+
+    const supportedPackages = Object.keys(packagesMap);
+    if (!supportedPackages.includes(frameworkPackage)) {
+      console.log('no supported package, skipping');
+      return null;
+    }
+
+    const builderInfo = {
+      name: getBuilder(builder),
+      options: main.getFieldValue(['core', 'builder', 'options']) || {},
+    } as const;
+
+    // TODO: once we have vite frameworks e.g. @storybook/react-vite, then we support it here
+    // and remove ['storybook-builder-vite', '@storybook/builder-vite'] from deps
+    if (builderInfo.name === 'vite') {
+      return null;
+    }
+
+    const frameworkOptions = getFrameworkOptions(framework, main);
+
+    const dependenciesToRemove = [
+      '@storybook/builder-webpack5',
+      '@storybook/manager-webpack5',
+    ].filter((dep) => allDeps[dep]);
+
+    const newFrameworkPackage = packagesMap[frameworkPackage][builderInfo.name];
+    const dependenciesToAdd = [];
+
+    // some frameworks didn't change e.g. Angular, Ember
+    if (newFrameworkPackage !== frameworkPackage) {
+      dependenciesToRemove.push(frameworkPackage);
+      dependenciesToAdd.push(newFrameworkPackage);
+    }
+
+    return {
+      main,
+      dependenciesToAdd,
+      dependenciesToRemove,
+      frameworkPackage: newFrameworkPackage,
+      frameworkOptions,
+      builderInfo,
+      packageJson,
+    };
+  },
+
+  prompt() {
+    return dedent`
+      We've detected you are using an older format of Storybook frameworks and builders.
+
+      In Storybook 7, frameworks also specify the builder to be used.
+
+      We can remove the dependencies that are no longer needed and install the new framework that already includes the builder.
+
+      More info: ${chalk.yellow(
+        'https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#framework-field-mandatory'
+      )}
+    `;
+  },
+
+  async run({
```

Comment:
Should we have a test for this or is it not something we can test right now? (need those E2E migration tests)

Reply:
Yes we definitely should, but it's too complicated right now however I do have plans on improving that in a near future